api/onix.py

```diff
@@ -0,0 +1,281 @@
+           
+            publishing_date_tag = parser._xpath(record, 'publishingdetail/publishingdate/b306')
+            issued = None
+            if publishing_date_tag:
+                date = publishing_date_tag[0].text
+                issued = datetime.datetime.strptime(date, "%Y%m%d")
+
+            identifier_tags = parser._xpath(record, 'productidentifier')
+            identifiers = []
+            primary_identifier = None
+            for tag in identifier_tags:
+                type = tag.find('b221').text
+                if type == '02' or type == '15':
+                    primary_identifier = IdentifierData(Identifier.ISBN, tag.find('b244').text)
+                    identifiers.append(primary_identifier)
+
+            subject_tags = parser._xpath(record, 'descriptivedetail/subject')
+            subjects = []
+            for tag in subject_tags:
+                type = tag.find('b067').text
+                if type in cls.SUBJECT_TYPES:
+                    subjects.append(SubjectData(cls.SUBJECT_TYPES[type],
+                                                tag.find('b069').text))
+                    
+            audience_tags = parser._xpath(record, 'descriptivedetail/audience/b204')
+            audiences = []
+            for tag in audience_tags:
+                if tag.text in cls.AUDIENCE_TYPES:
+                    subjects.append(SubjectData(Subject.FREEFORM_AUDIENCE,
+                                                cls.AUDIENCE_TYPES[tag.text]))
+
+            contributor_tags = parser._xpath(record, 'descriptivedetail/contributor')
+            contributors = []
+            for tag in contributor_tags:
+                type = tag.find('b035').text
+                if type in cls.CONTRIBUTOR_TYPES:
+                    display_name = tag.find('b036').text
+                    sort_name = tag.find('b037').text
+                    family_name = None
+                    family_name_tag = tag.find('b040')
+                    if family_name_tag != None:
+                        family_name = family_name_tag.text
+                    bio = None
+                    bio_tag = tag.find('b044')
+                    if bio_tag != None:
+                        bio = bio_tag.text
+                    contributors.append(ContributorData(sort_name=sort_name,
+                                                        display_name=display_name,
+                                                        family_name=family_name,
+                                                        roles=[cls.CONTRIBUTOR_TYPES[type]],
+                                                        biography=bio))
+
+            collateral_tags = parser._xpath(record, 'collateraldetail/textcontent')
+            links = []
+            for tag in collateral_tags:
+                type = tag.find('x426').text
+                # TODO: '03' is the summary in the example I'm testing, but that
+                # might not be generally true.
```

Comment:
I find the ONIX documentation difficult to use but if it's from "List 42" then "03" would be "Body matter", which is "Text components such as Part, Chapter, Section etc which appear as part of the main body of text content in a product", so that's not necessarily a summary.